### PR TITLE
Add colander_body_validator (fixes #386)

### LINKED ISSUE #386: Add a colander body validator ?
With the new validation API in #376, it is now possible to have schemas that validate things accross different parts of the request (e.g. querystring/headers/body). This is very convenient and powerful but can be tedious if you only need to validate body payloads because you have to define schemas like this:

```py

    from cornice.validators import colander_validator

    class Payload(colander.MappingSchema):
        username = colander.SchemaNode(colander.String())

    class SignupSchema(colander.MappingSchema):
        body = Payload()
        
    signup = cornice.Service()

    @signup.get(schema=SignupSchema, validators=(colander_validator,))
    def signup_get(request):
        username = request.validated['body']['username']
        return {'success': True}
```

We could have a `colander_body_validator` that validates the body only, which would prevent having to define a upper level schema:

```py

    from cornice.validators import colander_body_validator

    class SignupSchema(colander.MappingSchema):
        username = colander.SchemaNode(colander.String())
        
    signup = cornice.Service()

    @signup.get(schema=SignupSchema, validators=(colander_body_validator,))
    def signup_get(request):
        username = request.validated['username']  # no ['body']
        return {'success': True}
```

The code is short and would look like this:

```diff
diff --git a/cornice/validators/_colander.py b/cornice/validators/_colander.py
index 5136dca..ba83424 100644
--- a/cornice/validators/_colander.py
+++ b/cornice/validators/_colander.py
@@ -1,12 +1,28 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
-from colander import Invalid
+import colander
+
+from cornice.validators import extract_cstruct
+
+
+def _body_deserializer(request):
+    request_data = extract_cstruct(request)
+    return request_data['body']
+
+
+def body_validator(request, **kwargs):
+    schema = kwargs.get('schema')
+    if schema is not None:
+        request_schema = colander.MappingSchema()
+        request_schema.add('body', schema())
+        kwargs['schema'] = request_schema
+        kwargs['deserializer'] = _body_deserializer
+    return validator(request, **kwargs)
 
 
 def validator(request, deserializer=None, **kw):
     if deserializer is None:
-        from cornice.validators import extract_cstruct
         deserializer = extract_cstruct
 
     schema = kw.get('schema')
@@ -19,7 +35,7 @@ def validator(request, deserializer=None, **kw):
     try:
         deserialized = schema.deserialize(cstruct)
         request.validated.update(deserialized)
-    except Invalid as e:
+    except colander.Invalid as e:
         translate = request.localizer.translate
         error_dict = e.asdict(translate=translate)
         for name, msg in error_dict.items():
```

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim r?

### CODE REVIEW

File: cornice/tests/validationapp.py [hunk 1]
```diff
@@ -155,13 +155,23 @@ def post7(request):
...
     )
 
-    from cornice.validators import colander_validator
+    from cornice.validators import colander_validator, colander_body_validator
 
     COLANDER = True
 except ImportError:
     COLANDER = False
 
 if COLANDER:
```
Comment (leplatrem) [inline cornice/tests/validationapp.py]: `<msg mode="getshitdone">` Please do it in another PR, this is out of scope here `</msg>`

Comment (Natim) [inline cornice/tests/validationapp.py]: Yes

File: cornice/validators/__init__.py [hunk 2]
```diff
@@ -2,7 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 from webob.multidict import MultiDict
-from cornice.validators._colander import validator as colander_validator  # noqa
+from cornice.validators._colander import (
+    validator as colander_validator,
+    body_validator as colander_body_validator)
+
+
+__all__ = ['colander_validator', 'colander_body_validator', 'extract_cstruct']
```
Comment (Natim) [inline cornice/validators/__init__.py]: We probably want to add `DEFAULT_VALIDATORS` and `DEFAULT_FILTERS` too.

File: cornice/tests/validationapp.py (hunk 1 above)
Comment (Natim) [inline cornice/tests/validationapp.py]: Actually I was wrong, we are defining stuff there that needs colander to be defined.
Please ignore my comment.
